test/integration/http/http.go

```diff
@@ -69,33 +69,84 @@ var httpbinDep *appsv1.Deployment = &appsv1.Deployment{
+		Kind:       "Deployment",
+	},
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "nghttp2",
+	},
+	Spec: appsv1.DeploymentSpec{
+		Replicas: int32Ptr(1),
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{"application": "nghttp2"},
+		},
+		Template: apiv1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"application": "nghttp2",
+				},
+			},
+			Spec: apiv1.PodSpec{
+				Containers: []apiv1.Container{
+					{
+						Name:            "nghttp2",
+						Image:           "docker.io/svagi/nghttp2",
+						ImagePullPolicy: apiv1.PullIfNotPresent,
+						Ports: []apiv1.ContainerPort{
+							{
+								Name:          "nghttp2",
+								Protocol:      apiv1.ProtocolTCP,
+								ContainerPort: 8443,
+							},
+						},
+						//docker run -p 8443:8443 --network my-bridge-network -it svagi/nghttp2 nghttpx  -f"0.0.0.0,8443;no-tls" -b172.18.0.2,80 -L INFO
+						Command: []string{
+							"nghttpx",
+							"-f0.0.0.0,8443;no-tls",
+							"-bhttpbin,8080",
+							"-L",
+							"INFO",
+						},
+					},
+				},
+			},
+		},
+	},
 }
 
 func (r *HttpClusterTestRunner) RunTests(ctx context.Context, t *testing.T) {
-	//TODO https://github.com/skupperproject/skupper/issues/95
-	//all this hardcoded sleeps must be fixed, probably along with #95
-	//for now I am just keeping them in the same values that we are using
-	//for tcp_echo test, since in case of reducing test may fail
-	//intermitently
 	pubCluster1, err := r.GetPublicContext(1)
 	assert.Assert(t, err)
 
 	_, err = k8s.WaitForSkupperServiceToBeCreatedAndReadyToUse(pubCluster1.Namespace, pubCluster1.VanClient.KubeClient, "httpbin")
 	assert.Assert(t, err)
 
-	jobName := "http"
-	jobCmd := []string{"/app/http_test", "-test.run", "Job"}
-
-	_, err = k8s.CreateTestJob(pubCluster1.Namespace, pubCluster1.VanClient.KubeClient, jobName, jobCmd)
+	_, err = k8s.WaitForSkupperServiceToBeCreatedAndReadyToUse(pubCluster1.Namespace, pubCluster1.VanClient.KubeClient, "nghttp2")
 	assert.Assert(t, err)
 
-	job, err := k8s.WaitForJob(pubCluster1.Namespace, pubCluster1.VanClient.KubeClient, jobName, constants.ImagePullingAndResourceCreationTimeout)
-	assert.Assert(t, err)
-	pubCluster1.KubectlExec("logs job/" + jobName)
+	runJob := func(cc *base.ClusterContext, jobName, testName string) {
+		t.Helper()
+		jobCmd := []string{"/app/http_test", "-test.run", testName}
+
+		_, err = k8s.CreateTestJob(cc.Namespace, cc.VanClient.KubeClient, jobName, jobCmd)
+		assert.Assert(t, err)
+	}
 
-	k8s.AssertJob(t, job)
+	waitJob := func(cc *base.ClusterContext, jobName string) {
+		t.Helper()
+		job, err := k8s.WaitForJob(cc.Namespace, cc.VanClient.KubeClient, jobName, constants.ImagePullingAndResourceCreationTimeout)
+		assert.Assert(t, err)
+		cc.KubectlExec("logs job/" + jobName)
+		k8s.AssertJob(t, job)
+	}
+
+	runJob(pubCluster1, "http1", "TestHttpJob")
+	waitJob(pubCluster1, "http1")
+
+	runJob(pubCluster1, "http2", "TestHttp2Job")
+	waitJob(pubCluster1, "http2")
```

Comment:
@nicob87 test is passing locally for me, but I believe you could add a sub-test here,
when running http1 and http2 jobs. Something like:
```
	t.Run("http1", func(t *testing.T) {
		runJob(pubCluster1, "http1", "TestHttpJob")
		waitJob(pubCluster1, "http1")
	})

	t.Run("http2", func(t *testing.T) {
		runJob(pubCluster1, "http2", "TestHttp2Job")
		waitJob(pubCluster1, "http2")
	})
```

The benefit is that the generated output (and eventual junit.xml) will have results
defined separately for each http1 and http2 tests.

Reply:
makes totally sense